thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js

```diff
@@ -0,0 +1,252 @@
+/**
+ * Controller for Alert Landing and Details Page
+ * @module manage/alert
+ * @exports manage/alert
+ */
+import fetch from 'fetch';
+import Ember from 'ember';
+import moment from 'moment';
+import { checkStatus } from 'thirdeye-frontend/helpers/utils';
+import { task, timeout } from 'ember-concurrency';
+
+export default Ember.Controller.extend({
+  /**
+   * Set up to receive prompt to trigger page mode change.
+   * When replay is received as true, it indicates that this is a
+   * newly created alert and replay is needed in order to display anomaly data.
+   */
+  queryParams: ['replay'],
+  replay: false,
+
+  /**
+   * Standard legend settings for graph
+   */
+  legendText: {
+    dotted: {
+      text: 'WoW'
+    },
+    solid: {
+      text: 'Observed'
+    }
+  },
+
+  /**
+   * Set initial view values
+   * @method initialize
+   * @param {Boolean} isReplayNeeded
+   * @return {undefined}
+   */
+  initialize(isReplayNeeded) {
+    this.setProperties({
+      filters: {},
+      graphConfig: {},
+      selectedFilters: JSON.stringify({}),
+      isAlertReady: false,
+      isReplayStarted: true,
+      isReplayPending: false,
+      isReplayDone: false,
+      isReplayStatusError: false,
+      isReplayModeWrapper: true,
+      requestCanContinue: true,
+      selectedDimension: 'All Dimensions',
+      dateRangeToRender: [30, 10, 5],
+    });
+    // Toggle page mode if replay is needed
+    if (isReplayNeeded) {
+      this.set('isReplayPending', true);
+      this.triggerReplay(this.get('alertId'));
+    }
+  },
+
+  /**
+   * Generate date range selection options if needed
+   * @method renderDate
+   * @param {Number} range - number of days (duration)
+   * @return {String}
+   */
+  renderDate: function(range) {
+    // TODO: enable single day range
+    const newDate = moment().subtract(range, 'days').endOf('day').utc().format("DD MM YYY");
+    return `Last ${range} Days (${newDate} to Today)`;
+  },
+
+  /**
+   * Array containing the types of anomaly responses supported
+   * @type {Array}
+   */
+  responseOptions: ['Not reviewed yet', 'True anomaly', 'False alarm', 'I don\'t know', 'Confirmed - New Trend'],
```

Comment:
I think it's more readable if we have the variables defined at the top before methods and computed properties. Let me know what you think.